samples/boards/nrf52/mesh/onoff_level_app/src/mesh/process_message.c

```diff
@@ -0,0 +1,172 @@
+#include "common.h"
+
+void process_message(struct bt_mesh_model *model, struct bt_mesh_msg_ctx *ctx,
+		     struct net_buf_simple *buf, uint16_t opcode)
+{
+	uint8_t tid, tmp8;
+	int16_t tmp16;
+	s32_t err = 0, tmp32;
+	struct net_buf_simple *msg;
+	struct server_state *state = model->user_data;
+
+	switch (opcode) {
+	case 0x8201:	/* GEN_ONOFF_SRV_GET */
```

Comment:
I took a closer look at this function now (previously named "doer") why are you de-multiplex the message handlers into this single function, when the various case branches have essentially nothing in common (except perhaps some function variables). I'd either split this up to one function per message, or even better just merge the code back to the message handlers that call this function. Why was there a need to do this outside of the message handlers in the first place? (if there is a well justified need I'd still split this up into one function per message, and then you also solve the trouble you had with naming, since you can name the functions based on the message that they handle).